src/cloudsc_python/src/cloudsc4py/framework/storage.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+from contextlib import contextmanager
+import numpy as np
+from typing import TYPE_CHECKING
+
+import gt4py
+from sympl._core.data_array import DataArray
+
+from cloudsc4py.framework.grid import get_mask
+
+if TYPE_CHECKING:
+    from typing import Literal, Optional
+
+    from cloudsc4py.framework.config import GT4PyConfig
+    from cloudsc4py.framework.grid import ComputationalGrid, DimSymbol
+
+
+def zeros(
+    computational_grid: ComputationalGrid,
+    grid_id: tuple[DimSymbol, ...],
+    data_shape: Optional[tuple[int, ...]] = None,
+    *,
+    gt4py_config: GT4PyConfig,
+    dtype: Literal["bool", "float", "int"],
+) -> gt4py.storage.Storage:
+    grid = computational_grid.grids[grid_id]
+    data_shape = data_shape or ()
+    shape = grid.storage_shape + data_shape
+    dtype = gt4py_config.dtypes.dict()[dtype]
+    return gt4py.storage.zeros(shape, dtype, backend=gt4py_config.backend)
```

Comment:
I'm still getting an `AttributeError` here for GPU-storage backends. This is likely due to my borked gt4py install, but could we possibly catch this a bit more elegantly here? 

for reference:
```
  File "/lus/h2resw01/scratch/naml/dwarf-p-cloudsc/src/cloudsc_python/src/cloudsc4py/framework/storage.py", line 31, in zeros
    return gt4py.storage.zeros(shape, dtype, backend=gt4py_config.backend)
  File "/lus/h2resw01/scratch/naml/gt4py/src/gt4py/storage/storage.py", line 258, in zeros
    storage = empty(
  File "/lus/h2resw01/scratch/naml/gt4py/src/gt4py/storage/storage.py", line 106, in empty
    _, res = allocate_f(shape, layout_map, dtype, alignment * dtype.itemsize, aligned_index)
  File "/lus/h2resw01/scratch/naml/gt4py/src/gt4py/storage/utils.py", line 219, in allocate_gpu
    device_raw_buffer = cp.empty((buffer_size,), dtype=dtype)
AttributeError: 'NoneType' object has no attribute 'empty'
```